MyNumSetLibrary/MyNumSetLibrary/NumSet.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+
+namespace MyNumSetLibrary
+{
+    public class NumSet
+    {
+        private readonly HashSet<int> _HashSet;
+
+        public NumSet(params int[] hashSet)
+        {
+            _HashSet = new HashSet<int>(hashSet);
+        }
+
+        public override string ToString()
+        {
+            return string.Join(",", _HashSet);
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj is not NumSet)
+            {
+                return false;
+            }
+
+            return _HashSet.SetEquals(((NumSet)obj)._HashSet);
+        }
+
+        public override int GetHashCode()
+        {
+            int hashCode = 0;
+
+            foreach (int i in _HashSet)
+            {
+                hashCode += i;
+            }
+
+            return hashCode + _HashSet.Count;
```

Comment:
This works, but in newer version of .NET there is a convenient [HashCode.Combine](https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=net-5.0). The advantage of this is you then don't need to worry about the distribution of the hash code, or integer overflow;

Reply:
Are you talking about replacing the method with:
```
return HashCode.Combine(_HashSet);
```
How will I know the expected value?

Reply:
More like this: https://github.com/IntelliTect-Samples/EWU-CSCD371-2021-Winter/blob/Assignment4Complete/Assignment4/Assignment4/NumSet.cs#L42-L47